src/transformers/models/bart/modeling_bart.py

```diff
@@ -941,11 +929,21 @@ def forward(
             attention_mask = input_ids.ne(self.config.pad_token_id).to(torch.long)
             # never mask leading token, even if it is pad
             attention_mask[:, 0] = attention_mask[:, 1]
+            if past_key_values_length > 0:
```

Comment:
`past_key_values_length` is actually not the responsibility of the `expand_mask` function but of the attention mask -> cleaner to do it here.